zenodo/modules/thumbnails/cache.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2019 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Implements a Redis cache."""
+
+from __future__ import absolute_import
+
+from flask_iiif.cache.redis import ImageRedisCache as ImageCache
+
+
+class ImageRedisCache(ImageCache):
+    """Redis image cache."""
+
+    def __init__(self):
+        """Initialize the cache."""
+        super(ImageRedisCache, self).__init__()
+
+    def set(self, key, value, timeout=None):
+        """Cache the object.
+
+        :param key: the object's key
+        :param value: the stored object
+        :type value: `BytesIO` object
+        :param timeout: the cache timeout in seconds
+        """
+        identifier, _, size, _, _ = key.split('/')
+        if size == '250,':
```

Comment:
I think we should be checking for very specific parameters (i.e. `.../{full}/{250,}/{0}/{default}.{png}`).

E.g. this one would match even rotated images and cache them